documentation/disaster-recovery-testing.md

```diff
@@ -0,0 +1,176 @@
+
+If the selcted env has statuscake monitoring, then pause monitoring for the duration of the testing
+
+## Enable maintenance mode
+
+Run the enable maintenance workflow for the service and environment affected
+
+##
+## Follow instructions for the selected scenario
+##
+
+## Scenario 1. Loss of database instance
+
+In case the database instance is lost, the objectives are:
+
+- Recreate the lost postgres database instance
+- Restore data from nightly backup stored in Azure. The point-in-time and snapshot backups created by the Azure Postgres service will not be available if it's been deleted.
+
+### Delete the postgres database instance
+
+Note that you must have a previously created backup on azure storage before starting this step. If not, create one now before continuing.
+
+- Delete the existing postgres database
+    - manually delete via UI https://portal.azure.com/#browse/Microsoft.DBforPostgreSQL%2FflexibleServers
+- confirm it's deleted
+
+If deleting any other resources (scenario 3) then this should be done at this point too.
+
+### Recreate the lost postgres database instance
+
+Run the deploy workflow to recreate the missing postgres database (and any other missing resources).
+
+As the maintenance page has been implemented, you will need to
+- create a branch from main
+- update the env tfvars.json as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md#configure-terraform-to-keep-deploying-the-application
+- push the branch to github
+- run the deploy workflow using your branch for the env being tested
+
+If the web app fails to start due to the postgres db missing data
+- update your PR to override the start command to remove any db commands or a dummy command (e.g. sleep 999999)
+- push branch and rerun deploy workflow as above
+
+### Restore the data from previous backup in Azure atorage
+
+Run the restore from backup workflow
+
+### Redeploy app if required
+
+If the start command was updated previously
+- put back the original start command in your PR
+- push to github and rerun deploy
+
+### Validate app
+
+Confirm the app is working and can see the restored data
+
+
+## Scenario 2. Loss of data
+
+In the case of data loss, we need to recover the data as soon as possible in order to resume normal service.
+
+The application's database is an Azure flexible postgres instance. This provides a point-in-time restore (PTR) ability with the resolution of 1 second, available between 5min and 7days. PTR allows you to restore the source server to a point-in-time on a new copy of the server, not to the source server itself.
+
+The objectives of this test are:
+- Create a separate new postgres database instance
+- Restore data from the current live database to the new postgres database instance from a particular point in time
+- data can then be updated into the source database from the new PTR server
+    - do we want to test this as part of the DR test?
```

Comment:
Yes this would make sense as this is the most likely scenario

Reply:
updated